compiler/rustc_codegen_llvm/src/gotoc/mir_to_goto/codegen/typ.rs

```diff
@@ -990,7 +990,18 @@ impl<'tcx> GotocCtx<'tcx> {
         variants
             .iter()
             .filter_map(|lo| {
-                if lo.fields.count() == 0 { None } else { Some(lo.fields.offset(0).bits_usize()) }
+                if lo.fields.count() == 0 {
+                    None
+                } else {
+                    // get the offset of the leftmost field, which is the one
+                    // with the least offset since we codegen fields in a struct
+                    // in the order of increasing offsets
```

Comment:
Note this is not necessarily the 0th field, since the compiler may reorder fields.